### PR TITLE
Use vtk vector mapping from vtkScalarstoColors in add_mesh

### PR DESCRIPTION
**Pull request by MatthewFlamm**
### Overview

Instead of modifying vector arrays for plotting in add_mesh, this PR now uses the vtk functionality to map vectors through [vtkScalarstoColors](https://vtk.org/doc/nightly/html/classvtkScalarsToColors.html).  The use of this functionality has some beneifts:

1) Defers the logic of mapping to vtk
2) Plotting multiple components no longer requires copying the whole mesh

I tried to keep this PR as a non-breaking change.  I'm still not sure whether `scalars.ravel()` makes sense when we have data with more than 2 dimensions, but I left that functionality intact.

### Details

I'd like to implement this in another PR for `add_volume` in a shared helper routine, but this will be a breaking change as `add_volume` currently does `scalars.ravel()` for vector data. 

The image regression check was originally flawed in that the clim scale was reset for each component, but should now work properly.  I committed the new image.

Somewhat related to #1321, #1319, and https://github.com/pyvista/pyvista-support/issues/425.

In https://github.com/pyvista/pyvista/issues/1319#issuecomment-842724958, it was asked 

> Does pyvista even support non-3d vectors/tensors?

Using this PR, I tested 4D and 5D vectors, and they work using this functionality.  I suspect in other areas like streamlines, it wouldn't work, but I haven't tried.

### CODE REVIEW

**Comment by MatthewFlamm** [comment]
The failure is related to passing data to `ipygany.Component` in building the documentation. [Failure link](https://dev.azure.com/pyvista/PyVista/_build/results?buildId=6072&view=logs&j=0d0124fa-d3a8-51e8-d2f8-a0118e38d729&t=6685bac6-5dce-56af-10d7-64cc75251afe&l=789). It happens on line 81 since array cannot take a 2D array and it seems `ipygany.Component` requires 1 component at a time.
https://github.com/pyvista/pyvista/blob/cc22f1f75e9d6a3b58a5b5b6a08d83a1623c1ac5/pyvista/jupyter/pv_ipygany.py#L78-L83

I will need to dig into `ipygany` to understand whether this can be supported.  At first blush, it seems straightforward to simply append multiple components if the data array is 2 dimensional.  But, we will also need to support passing which component is to be plotted somewhere.  I think that happens here:

https://github.com/pyvista/pyvista/blob/cc22f1f75e9d6a3b58a5b5b6a08d83a1623c1ac5/pyvista/jupyter/pv_ipygany.py#L155

**Comment by MatthewFlamm** [comment]
I'm hoping this is now fixed.  I was able to add a test, but I wasn't able to directly test the documentation example.

I added functionality to pass all components of vector data to ipygany.  ipygany does not support taking the magnitude of the data, as far as I can tell.  For now, I've raised an error in pyvista if this is requested.  Note that this will be a breaking change from the current behavior.  In my opinion, it would be better for ipygany to support plotting the magnitude of multi-component data and I could raise an issue request there.

If we do want to support it directly in pyvista now, we could pass the magnitude of the data to ipygany as an additional component, i.e. a 3D vector would become a 4D vector with the 4th dimension being the magnitude.  Then we could still plot the magnitude.  This would be a hack to get around limitations in ipygany.

**Comment by akaszynski** [comment]
> Note that this will be a breaking change from the current behavior. In my opinion, it would be better for ipygany to support plotting the magnitude of multi-component data and I could raise an issue request there.

Agreed, however think `ipygany` development has been stalled since their last commit was in March.

> If we do want to support it directly in pyvista now, we could pass the magnitude of the data to ipygany as an additional component, i.e. a 3D vector would become a 4D vector with the 4th dimension being the magnitude. Then we could still plot the magnitude. This would be a hack to get around limitations in ipygany.

This might be necessary if we want to continue using `ipygany`.  I'm still waiting for a clear jupyterlab 3d visualization library to emerge that we can stake our development to, but right now it's still quite fractured.

Let's implement that in a separate PR.  For now, this is a great way of cleaning up component plotting without resorting to the `copy` hack.

File: tests/plotting/test_plotting.py [hunk 1]
```diff
@@ -1058,19 +1058,23 @@ def test_vector_array():
...
-    p.add_mesh(data.copy(), scalars="vector_values_points", component=1)
+    p.add_mesh(data, scalars="vector_values_points", component=1, **kwargs)
     p.subplot(1, 1)
-    p.add_mesh(data.copy(), scalars="vector_values_points", component=2)
+    p.add_mesh(data, scalars="vector_values_points", component=2, **kwargs)
     p.link_views()
-    p.show()
 
-    # p.show(before_close_callback=verify_cache_image)
+    p.show(before_close_callback=verify_cache_image)
```
Comment (akaszynski) [inline tests/plotting/test_plotting.py:1077]: Nice work adding this back in.

**Review by akaszynski — APPROVED** [review]
LGTM.  Thanks for adding the image verification back in.

**Comment by MatthewFlamm** [comment]
Thanks.  I'll let this cure for a few days in case there are any other comments before merging.

I raised an issue in ipygany for using the magnitude of multicomponent data.  If there is no movement, we'll add in a new PR here.

**Comment by akaszynski** [comment]
> Thanks. I'll let this cure for a few days in case there are any other comments before merging.

Hope you were fine with me merging.

**Pull request merged**